app.py

```diff
@@ -0,0 +1,369 @@
+@jwt_required()
+def delete_book_details(book_id):
+    user_id = get_jwt_identity()
+    book= Books.query.filter(db.and_(Books.profile_id==user_id , Books.id==book_id)).first()
+    if book:
+        delete(book)
+        return jsonify({'message': 'Book deleted'})
+    else:
+        return jsonify({'error':'Book is not added by you.'})
+
+
+@app.route('/books', methods=['GET'])
+def get_books(book_id=None):
+    data=request.json
+    book_id=data.get('book_id')
+    if book_id is None:
+
+       all_books = Books.query.all()
+       book_list = []
+       for book in all_books:
+           book_list.append({
+            'id': book.id,
+            'title': book.title,
+        })
+       return jsonify(book_list)
+  
+    
+    book = Books.query.get(book_id)
+    if book:
+        return jsonify({
+            'id': book.id,
+            'title': book.title,
+            'author': book.author,
+            'isbn': book.isbn,
+            'genre': book.genre,
+            'publication_year': book.publication_year
+        })
+    else:
+        return jsonify({'error': 'Book not found'})
+    
+
+
+    """
+
+
+
+    Author Mangemnt
+    
+    
+    """
+
+
+@app.route('/author/add', methods=['POST'])
+@jwt_required()
+def add_author():
+    user_id = get_jwt_identity()
+    data = request.json
+    author_name = data.get('author_name')
+    biography = data.get('biography')
+    nationality = data.get('nationality') 
+    if  author_name and biography:
+         if Author.query.filter(db.and_(Author.profile_id==user_id , Author.author_name==author_name , Author.biography==biography)).first():
+             return jsonify({'error':'Author is already added by you.'})
+         else:
+             
+            new_author = Author(author_name = author_name, biography = biography,nationality = nationality
+                         ,profile_id=user_id)
+            new_add(new_author)
+            
+            return jsonify({'message': 'New book added'})
+    else:
+        return jsonify({'error': 'Provide valid fields' })
+    
+
+
+
+@app.route('/author', methods=['GET'])
+def get_author(author_id=None):
+    data=request.json
+    author_id=data.get('author_id')
+    if author_id is None:
+        all_authors = Author.query.all()
+        author_list = []
+        for author in all_authors:
+            author_list.append({
+            'id': author.id,
+            'author_name': author.author_name,
+        })
+        return jsonify(author_list)
+    
+
+    author_details = Author.query.get(author_id)
+    if author_details:
+            return jsonify({
+            'id': author_details.id,
+            'author_name': author_details.author_name,
+            'biography': author_details.biography,
+            'nationality': author_details.nationality,
+        })
+    else:
+         return jsonify({'error': 'Author not found'})
+
+
+@app.route('/author/delete/<int:author_id>', methods=['DELETE'])
```

Comment:
This API deletes the selected author using the author ID. We also need to add the logic to delete all authors if the author ID is none.